wordpress-mobile/WordPress-iOS
# Stats: Create missing ghost (loading) views

Fixes #23059

Created missing ghost loading views:
- For 3-column cell
- For a single value cell
- For a line chart

### Solution

We use Ghostable extension implemented in `WordPressSharedUI` which animates the top leaf views. 

- Improved `StatsGhostTopCell` to support dynamic input of the "number of columns".
- Added `StatsGhostSingleValueCell` to support ghost loading of single value cells, such as Total Subscribers
- Created `StatsGhostLineChartCell` to create a dummy line chart path and mask ghostable layer

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/038f82fd-9532-4f08-9140-7bf96d99c133

### To test:

For easier testing, set a slow `Network Link Conditioner` setting.

1. Open Stats -> Subscribers
2. Observe new ghostable loading views:
- Single value for Total Subscribers
- Line chart for Subsribers chart
- 2 columns for Subscribers list
- 3 columns for Emails
3. Change orientation and switch between dark/light mode and confirm the view adapts

## Regression Notes
1. Potential unintended areas of impact

Places that use `StatsGhostTopCell` should continue working as before, since I set 2 column scenario as default

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


Testing checklist:
